artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/RefsOperation.java

```diff
@@ -190,9 +190,7 @@ private void decrementRefCount(MessageReference refmsg) {
 
    @Override
    public synchronized List<MessageReference> getRelatedMessageReferences() {
-      List<MessageReference> listRet = new LinkedList<>();
```

Comment:
I'm not sure this is a good idea. I think this needs to be isolated.

at least that would break synchronization on the list.. a method listing JSon while the transaction still being appeneded.. you could get consequences almost impossible to test.


Can you close this PR pretty please?


I appreciate you know the codebase well... but this was meant to be

Reply:
I think the point here is the code is addall of the list it just created. This seems very odd. 

Id agree an internal list should just be exposed. But the current logic just looks crazy wrong. Youd be better to return Collections.emptylist!

Reply:
Sure thing @clebertsuconic, I appreciate the explanation and I'll close that one.

Reply:
Yeah, the original commit was just replacing the addAll with a return of an empty list. I modified it to return refsAcks. 

Given that there can be unintended consequences here, I am closing the PR and letting those of you with more experience on the code base to decide what may be the more appropriate solution in this case.

Nonetheless, I agree with @michaelandrepearce, that there is something strange with the original code that the PR intended to fix.

Reply:
I guess I don't understand the issue then... /me back to looking into this.

Reply:
OHHHHHH... I see now...


I was reading the method as it was supposed to be written:


```java

      List<MessageReference> listRet = new LinkedList<>();
      listRet.addAll(refsToAck);
      return listRet;
```


@orpiske  this is how this method was supposed to be written ^^

Reply:
when I read addAll, my brain was tricking me and I was reading refsToAck...


if we make that change... it will be ok...


I guess this needs to validate if empty... or null and be a proper change.

Reply:
@clebertsuconic thanks for taking the time to go through the code and checking how it should be. I will reopen this PR and do the change as instructed!